sdk-lib/src/main/java/cash/z/ecc/android/sdk/block/CompactBlockProcessor.kt

```diff
@@ -618,8 +619,9 @@ class CompactBlockProcessor internal constructor(
             twig("Found UTXO at height ${utxo.height.toInt()} with ${utxo.valueZat} zatoshi")
             @Suppress("TooGenericExceptionCaught")
             try {
+
                 rustBackend.putUtxo(
```

Comment:
Great that you noticed this.  I think checking the return value is important, and we should at a minimum log the failure.

Failure writing to the Rust backend should probably be consistent (it's writing to the SQLite database which isn't expected to be flaky as far as I know).  Will a retry improve reliability in this particular case?  I suspect the retries are important for the network downloads, but maybe not the database writes.

@str4d Would a failure writing to the rust backend here succeed on a second try?

Reply:
I'm also thinking about why do we return `true/false` as well as throwing error in the rust layer? Could this be unified, or is it a necessary pattern here? 

Or may be another question, is the Rust really throwing an exception and we should `try-catch` it? Because I can see that the Rust does not throw exceptions as we know them from the Android world, it just returns `Result<T, E>` or use `panic`, which is probably the reason why we have to have the `try-catch` here, right? Thank you @str4d.

Reply:
Thank you for this explanation.

A quick solution might be: tweak many of the RustBackend public APIs to have void return values (leaving the `external fun` declarations along with the FFI code alone for a second refactor).  We'd need to add a comment in RustBackend explaining why the booleans are being ignored.  Then it would be contained within a single file.

Reply:
> Would a failure writing to the rust backend here succeed on a second try?

It would depend on the failure. At a glance, the kinds of errors that can occur from this function include:
- Passing an unknown network ID.
- Wallet DB path not existing or not being openable.
- SQLite errors of basically any kind, including violations of table constraints (which would likely re-occur on retry) and transient write failures (which might not).
- Attempting to store a UTXO that is not P2PKH or P2SH.

The non-SQLite errors occurring would indicate logic bugs on the Android side, so assuming those don't occur... the answer is "maybe"? For example, if the failure occurs because the UTXO already existed in the wallet DB (failing the `prevout` uniqueness constraint), due to e.g. a race condition on the Android side, that will continue to fail.

Reply:
Thank you @str4d for this information. We have two more things for you, see below. Could you please take a quick look at them too, once you have some spare time so we can have this merged? Thank you.

I removed the new boolean check of this method and filed a new issue for changing related RustBackend method return types to void, as discussed here. https://github.com/zcash/zcash-android-wallet-sdk/issues/920.

Reply:
Should #920 block this PR from merging then? Otherwise this should still check the boolean result until the alternative approach is implemented.

Reply:
If desired, we could add a comment noting issue 920 and that the return value is not useful in this context.

Reply:
Yep, comments added to have this clear for everyone. We're good without boolean checking here, as we previously already had the try-catch result checking approach. So the https://github.com/zcash/zcash-android-wallet-sdk/issues/920 is not blocking this PR.